chpl/chpl-service/src/main/java/gov/healthit/chpl/app/chartdata/NonconformityTypeChartCalculator.java

```diff
@@ -0,0 +1,65 @@
+package gov.healthit.chpl.app.chartdata;
+
+import java.util.List;
+
+import gov.healthit.chpl.dao.statistics.NonconformityTypeStatisticsDAO;
+import gov.healthit.chpl.dao.statistics.SurveillanceStatisticsDAO;
+import gov.healthit.chpl.dto.NonconformityTypeStatisticsDTO;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.orm.jpa.JpaTransactionManager;
+import org.springframework.transaction.TransactionStatus;
+import org.springframework.transaction.support.TransactionCallbackWithoutResult;
+import org.springframework.transaction.support.TransactionTemplate;
+
+public class NonconformityTypeChartCalculator {
+	
+	private static final Logger LOGGER = LogManager.getLogger(NonconformityTypeChartCalculator.class);
+	
+    private SurveillanceStatisticsDAO statisticsDAO;
+    private NonconformityTypeStatisticsDAO nonconformityTypeStatisticsDAO;
+    private JpaTransactionManager txnManager;
+    private TransactionTemplate txnTemplate;
+    
+    NonconformityTypeChartCalculator(final ChartDataApplicationEnvironment appEnvironment) {
+    	statisticsDAO = (SurveillanceStatisticsDAO) appEnvironment
+                .getSpringManagedObject("surveillanceStatisticsDAO");
+    	nonconformityTypeStatisticsDAO = (NonconformityTypeStatisticsDAO) appEnvironment
+                .getSpringManagedObject("nonconformityTypeStatisticsDAO");
+        txnManager = (JpaTransactionManager) appEnvironment.getSpringManagedObject("transactionManager");
+        txnTemplate = new TransactionTemplate(txnManager);
+    }
+    
+    NonconformityTypeChartCalculator(final SurveillanceStatisticsDAO statisticsDAO,
+    		NonconformityTypeStatisticsDAO nonconformityTypeStatisticsDAO,
+    		TransactionTemplate txnTemplate) {
+    	this.statisticsDAO = (SurveillanceStatisticsDAO) statisticsDAO;
+    	this.nonconformityTypeStatisticsDAO = nonconformityTypeStatisticsDAO;
+        this.txnTemplate = txnTemplate;
```

Comment:
More odd indentation stuff. Did you do checkstyle against all of the files you're editing? That's the best way I've found to ensure gradual improvement over time.

Also should run findbugs and see if there are any bugs you can fix in the code you're changing / adding. Don't have to fix ones outside this code, but shouldn't add new ones, and should burn down old ones.